### DISCUSSION TITLE
Is there a way to use streaming and await together?

### DISCUSSION THREAD

**Discussion by sbarski** [guidance-ai/guidance#177] [Q&A]
Hello, right now it seems that we can either use `stream=True` or the `await` command but not both together (I am on version 0.0.61).

The specific error when trying to pause execution and then resume a partially executed guidance program (**with streaming enabled**) is: 'generator' object is not callable error message. This is understandable as enabling the stream return a generator via a yield. 

In fact, the current example in the readme that uses `stream=True` and `await` doesn't work for me: 
https://github.com/microsoft/guidance#pausing-execution-with-await. I get the above error message. 

If I set `stream=False` though then it works fine. Perhaps I am making a simple mistake somewhere (I hope!); but if I am not, are there any recommendations for making streaming and await working together nicely?

PS. thanks for a great project!

**Comment by jaycelq** [comment]
@slundberg I encountered the same problem, in Stream mode, the executor exited directly without await.
